File: chatGPT35_qualif1.java
package com.isograd.exercise;
import java.util.*;

public class IsoContest {
    public static void main( String[] argv ) throws Exception {
        String line;
        Scanner sc = new Scanner(System.in);
        line = sc.nextLine();
        /* Lisez les données et effectuez votre traitement */
        int length = line.length();
        if (length == 1) {
            System.out.println(line);
        } else {
            char winner = line.charAt(0);
            for (int i = 1; i < length; i++) {
                char current = line.charAt(i);
                if (current == 'C' && winner == 'F') {
                    winner = current;
                } else if (current == 'F' && winner == 'P') {
                    winner = current;
                } else if (current == 'P' && winner == 'C') {
                    winner = current;
                }
            }
            System.out.println(winner);
        }
    }
}

File: chatGPT35_qualif1.py
import sys

lines = []
for line in sys.stdin:
    lines.append(line.rstrip('\n'))

# Lisez les données et effectuez votre traitement
line = lines[0]
length = len(line)

if length == 1:
    print(line)
else:
    winner = line[0]
    for i in range(1, length):
        current = line[i]
        if current == 'C' and winner == 'F':
            winner = current
        elif current == 'F' and winner == 'P':
            winner = current
        elif current == 'P' and winner == 'C':
            winner = current
    print(winner)
